### ISSUE TITLE
Torch VS TRT layer-wise debug ?

### ISSUE THREAD

**Issue by dedoogong** [NVIDIA-AI-IOT/torch2trt#310]
after a tons of trial and errors,
I finally struggled to have converted my model to TRT and run successfully.

But the result accuracy was little bit worse than my expectation( even it was just FP32 conversion ).

I guess there the causes would come from layer fusion. 

I would like to debug layer by layer, but for example some part of my layers are packed into a nn.Sequential(). 

networ.mark_output("each layers output tensor") would be the workaround, but I think I can't apply this method when I use nn.Sequential()

Would you please kindly give me any hint to debug all layers between torch vs TRT?

Thank you!

**Comment by ma-siddiqui** [comment]
In my case, the results are much more worse. I also struggled a lot to convert and run trt model successfully. I am also looking the same solution.

**Comment by jaybdub** [comment]
Hi All,

Thanks for reaching out!

Are you able to share the PyTorch model you're attempting to convert?

Best,
John

**Comment by dedoogong** [comment]
yes I converted the model from https://github.com/magicleap/SuperGluePretrainedNetwork

**Comment by dedoogong** [comment]
I hope to find a easier way to manually handle some layer not to be fused as an optional torch2trt argument ( such as each layer's unique id or name)

**Comment by jaybdub** [comment]
Hi dedoogong,

### Issue 1 - Incorrect results after loading model

Could you share the following information

1. Version of TensorRT
2. Version of PyTorch
3. Commit hash of torch2trt (or just branch name, probably "master")

Also, does your model use an interpolation / upsampling layer?  I've refactored the serialization of the interpolate plugin in this branch if you want to give it a try.  

https://github.com/NVIDIA-AI-IOT/torch2trt/pull/307

The refactoring is intended to resolve a different issue but would be worth checking.

### Issue 2 - Handling layer not to be fused

Do you mind sharing a small example of what you mean by a layer "not to be fused"?  Do you mean to skip adding the layer (an identity operation)?

**Comment by dedoogong** [comment]
Hello!

Issue 1 - Incorrect results after loading model
1. TRT 7 
2. Pytorch 1.3
3. master , built with plugin.
4. my model doesn't use interpolation at all. 

Issue 2 - Handling layer not to be fused
small example:

original model


"not to be fused" version (for being marked as outputs by TRT)

**Comment by poornimajd** [comment]
@dedoogong and @ma-siddiqui have you been  able to solve the issue?
